src/coprocessor/dag/expr/column.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::borrow::Cow;
+
+use coprocessor::codec::Datum;
+use coprocessor::codec::mysql::{Decimal, Duration, Json, Time};
+use super::{constant, Column, Error, Result};
+
+impl Column {
+    fn check_offset(&self, row: &[Datum]) -> Result<()> {
+        if self.offset >= row.len() {
+            return Err(Error::ColumnOffset(self.offset));
+        }
+        Ok(())
+    }
+
+    #[inline]
+    pub fn eval_int(&self, row: &[Datum]) -> Result<Option<i64>> {
+        try!(self.check_offset(row));
```

Comment:
1. Now we use `Expression::try_from` to build expression tree. Will `Expression::build_from(tipb.Expr, row_length)` be a better style?
2. TiKV doesn't store some `null` in value, for saving space. But after we parse it into `row`, we fill them back.